src/Kestrel.Core/Internal/AddressBinder.cs

```diff
@@ -131,7 +137,14 @@ private static Task BindEndpointAsync(IPEndPoint endpoint, AddressBindContext co
 
             try
             {
-                await BindEndpointAsync(new IPEndPoint(IPAddress.Loopback, address.Port), context).ConfigureAwait(false);
+                var options = new ListenOptions(new IPEndPoint(IPAddress.Loopback, address.Port));
+                await BindEndpointAsync(options, context).ConfigureAwait(false);
+                
+                if (https)
+                {
+                    options.KestrelServerOptions = context.ServerOptions;
+                    context.DefaultHttpsProvider?.ConfigureHttps(options);
```

Comment:
We want this to throw if DefaultHttpsProvider is null, no? You asked for https and you're not getting it.